python/examples/mlir/memref_management.py

```diff
@@ -0,0 +1,117 @@
+import torch
+import ctypes
+
+from mlir import ir
+from mlir.dialects import func, memref
+from mlir.runtime import np_to_memref
+from mlir.execution_engine import ExecutionEngine
+from mlir.passmanager import PassManager
+
+import lighthouse.utils as lh_utils
+
+
+def create_mlir_module(ctx: ir.Context, shape: list[int]) -> ir.Module:
+    with ctx, ir.Location.unknown():
```

Comment:
```suggestion
    
```
Note that `ir.Location` work similar to `ir.Context`. That is, you can just activate one in a context manager and relevant API pick it up automatically as well as it being possible to access it with `ir.Location.current` anywhere whilst the context manager has not exited (also in called functions).

So I would suggest entering a `with ir.Location.unknown()` as early as possible. When particular IR builders do have better location info, they can just enter another `with ir.Location.file(...)` for that particular scope (`ir.Location.current` is always the top of the stack of `ir.Location` context managers).